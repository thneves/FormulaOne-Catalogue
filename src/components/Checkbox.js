import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import '../styles/components/Checkbox.scss';

const Checkbox = ({
  handleChange, checked,
}) => {
  const list = useSelector(state => state.seasons.seasons);
  return list.map(year => (
    <div className="modal-checkbox" key={year}>
      <label htmlFor={year}>
        {year}
        <input type="checkbox" value={year} id={year} onChange={handleChange} defaultChecked={checked(year)} />
      </label>
    </div>
  ));
};

Checkbox.propTypes = {
  handleChange: PropTypes.func.isRequired,
  checked: PropTypes.func.isRequired,
};

export default Checkbox;
