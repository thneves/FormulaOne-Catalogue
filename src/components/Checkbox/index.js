import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

const Checkbox = ({
  handleChange, checked,
}) => {
  const list = useSelector(state => state.seasons.seasons);
  return list.map(year => (
    <div key={year}>
      <label htmlFor={year}>
        {year}
        <input type="checkbox" value={year} id={year} onChange={handleChange} defaultChecked={checked(year)} />
      </label>
    </div>
  ));
};

Checkbox.propTypes = {
  handleChange: PropTypes.func.isRequired,
  checked: PropTypes.func.isRequired,
};

export default Checkbox;
