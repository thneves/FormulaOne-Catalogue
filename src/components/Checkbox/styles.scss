.modal-checkbox {
  padding: 2rem;
  text-align: center;

  label {
    font-size: 1.3rem;
    font-weight: 700;
  }

  input {
    width: 1rem;
    height: 1rem;
  &:hover {
    cursor: pointer;
    }  
  }
}