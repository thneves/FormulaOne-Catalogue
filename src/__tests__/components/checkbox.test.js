import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from '../../components/Checkbox/index';
import * as redux from 'react-redux';

const useSelectorMock = jest.spyOn(redux, 'useSelector');

beforeEach( () => {
  useSelectorMock.mockClear();
})

it('should render Checkbox component', () => {

  useSelectorMock.mockReturnValue([2012, 2013, 2014])
  const checkbox = renderer
    .create(
      <Checkbox checked={() => true} handleChange={() => ''} />
    )

    expect(checkbox.toJSON()).toMatchSnapshot();
})