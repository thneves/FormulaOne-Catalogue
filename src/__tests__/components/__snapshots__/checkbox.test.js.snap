// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Checkbox component 1`] = `
Array [
  <div
    className="modal-checkbox"
  >
    <label
      htmlFor={2012}
    >
      2012
      <input
        defaultChecked={true}
        id={2012}
        onChange={[Function]}
        type="checkbox"
        value={2012}
      />
    </label>
  </div>,
  <div
    className="modal-checkbox"
  >
    <label
      htmlFor={2013}
    >
      2013
      <input
        defaultChecked={true}
        id={2013}
        onChange={[Function]}
        type="checkbox"
        value={2013}
      />
    </label>
  </div>,
  <div
    className="modal-checkbox"
  >
    <label
      htmlFor={2014}
    >
      2014
      <input
        defaultChecked={true}
        id={2014}
        onChange={[Function]}
        type="checkbox"
        value={2014}
      />
    </label>
  </div>,
]
`;
